common/flatpak-context.c

```diff
@@ -1182,8 +1187,12 @@ option_filesystem_cb (const gchar *option_name,
   FlatpakContext *context = data;
   g_autofree char *fs = NULL;
   FlatpakFilesystemMode mode;
+  gboolean explicit_suffix = FALSE;
+
+  if (!parse_filesystem_flags(value, FALSE, &mode, &explicit_suffix, error))
+      return FALSE;
```

Comment:
This is now parsing the argument twice, which is unnecessary. Instead, please pass the `explicit_suffix_out` argument through `flatpak_context_parse_filesystem`, the same way it already handles `mode_out`; and move your new check for `explicit_suffix` down below the `flatpak_context_parse_filesystem` call.

You'll need to change other callers of `flatpak_context_parse_filesystem` (in `common/flatpak-context.c` and `tests/test-exports.c`) to pass `NULL` for `explicit_suffix_out`, which means "ignore whether the suffix was explicitly given or not".